test/test_flop_counter.py

```diff
@@ -0,0 +1,123 @@
+# Owner(s): ["module: unknown"]
+
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+import torch.utils.flop_counter
+import unittest
+
+try:
+    from torchvision import models as torchvision_models
+    HAS_TORCHVISION = True
+except ImportError:
+    HAS_TORCHVISION = False
+skipIfNoTorchVision = unittest.skipIf(not HAS_TORCHVISION, "no torchvision")
+
+def FlopCounterMode(*args, **kwargs):
+    return torch.utils.flop_counter.FlopCounterMode(*args, **kwargs, display=False)
+
+def get_total_flops(mode):
+    return str(sum([v for _, v in mode.flop_counts["Global"].items()]))
+
+def T(*shape):
+    return torch.randn(*shape)
+
+class TestFlopCounter(TestCase):
+    def test_flop_counter_variety(self):
+        mode = FlopCounterMode()
+        mod = torch.nn.Linear(9, 10)
+        with mode:
+            torch.mm(T(4, 5), T(5, 6))
+            torch.matmul(T(5, 6), T(6, 7))
+            torch.einsum("ab,bc->ac", T(6, 7), T(7, 8))
+            mod(T(8, 9))
+
+        self.assertExpectedInline(get_total_flops(mode), """2852""")
+
+    def test_op(self):
+        mode = FlopCounterMode()
+        with mode:
+            torch.mm(T(4, 5), T(5, 6))
+        # 4 * 6 * 2 * 5 = 216
+        self.assertExpectedInline(get_total_flops(mode), """240""")
```

Comment:
Odd. 4 * 6 * 2 * 5 = 240. But the number of FLOPS for an `mm` `[m, k] x [k, n]` is `m * n * (2 * k - 1)`, which in this case gives 216.